pkg/linters/forgbiddenusage/forbiddenusage.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package forbiddenUsageprovides is a linter to detect unwanted/deprecated function calls in the code base.
+package forbiddenUsage
+
+import (
+	"go/ast"
+	"strings"
+
+	"github.com/golangci/plugin-module-register/register"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+func init() {
+	register.Plugin("forbiddenUsage", New)
+}
+
+type forbiddenUsagePlugin struct{}
+
+// New returns a new forbiddenUsage linter plugin
+func New(any) (register.LinterPlugin, error) {
+	return &forbiddenUsagePlugin{}, nil
+}
+
+// BuildAnalyzers returns the analyzers for the plugin
+func (f *forbiddenUsagePlugin) BuildAnalyzers() ([]*analysis.Analyzer, error) {
+	return []*analysis.Analyzer{
+		{
+			Name:     "forbiddenUsage",
+			Doc:      "detect unwanted/deprecated function calls",
+			Run:      f.run,
+			Requires: []*analysis.Analyzer{inspect.Analyzer},
+		},
+	}, nil
+}
+
+// GetLoadMode returns the load mode for the plugin
+func (f *forbiddenUsagePlugin) GetLoadMode() string {
+	return register.LoadModeSyntax
+}
+
+// Extend this list to detect more unwanted method call in the code base.
+var forbiddenUsage = map[string]string{
+	"github.com/DataDog/datadog-agent/pkg/config/model.NewConfig": "do not create new config, use 'pkg/config/mock' for test and the config comp otherwise",
+}
```

Comment:
Do you know if this could be defined in the golangci-lint config file directly ?
Similarly to `depguard` which defines a list of forbidden packages along with an explanation why they are forbidden

Reply:
I was going. to suggest something similar 😄 

I think it would allow to extend the linter with more ease.

golangci-lint allows to pass configuration information.

Reply:
Great, idea ! I'll do it in another PR.

Reply:
Actually it was simpler that I thought so I did it in this PR.